Application/EdFi.Ods.AdminApi/Features/OdsInstanceDerivative/AddOdsInstanceDerivative.cs

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using AutoMapper;
+using EdFi.Ods.AdminApi.Helpers;
+using EdFi.Ods.AdminApi.Infrastructure;
+using EdFi.Ods.AdminApi.Infrastructure.Database.Commands;
+using EdFi.Ods.AdminApi.Infrastructure.Database.Queries;
+using EdFi.Ods.AdminApi.Infrastructure.Helpers;
+using FluentValidation;
+using Microsoft.Extensions.Options;
+using Swashbuckle.AspNetCore.Annotations;
+
+namespace EdFi.Ods.AdminApi.Features.OdsInstanceDerivative;
+
+public class AddOdsInstanceDerivative : IFeature
+{
+    public void MapEndpoints(IEndpointRouteBuilder endpoints)
+    {
+        AdminApiEndpointBuilder
+           .MapPost(endpoints, "/odsInstanceDerivatives", Handle)
+           .WithDefaultDescription()
+           .WithRouteOptions(b => b.WithResponseCode(201))
+           .BuildForVersions(AdminApiVersions.V2);
+    }
+
+    public async Task<IResult> Handle(Validator validator, IAddOdsInstanceDerivativeCommand addOdsInstanceDerivativeCommand, IMapper mapper, AddOdsInstanceDerivativeRequest request)
+    {
+        await validator.GuardAsync(request);
+        var addedOdsInstanceDerivative = addOdsInstanceDerivativeCommand.Execute(request);
+        return Results.Created($"/odsInstanceDerivatives/{addedOdsInstanceDerivative.OdsInstanceDerivativeId}", null);
+    }
+
+  
+    [SwaggerSchema(Title = "AddOdsInstanceDerivativeRequest")]
+    public class AddOdsInstanceDerivativeRequest : IAddOdsInstanceDerivativeModel
+    {
+        [SwaggerSchema(Description = FeatureConstants.OdsInstanceDerivativeOdsInstanceIdDescription, Nullable = false)]
+        public int OdsInstanceId { get; set; }
+        [SwaggerSchema(Description = FeatureConstants.OdsInstanceDerivativeDerivativeTypeDescription, Nullable = false)]
+        public string? DerivativeType { get; set; }
+        [SwaggerSchema(Description = FeatureConstants.OdsInstanceDerivativeConnectionStringDescription, Nullable = false)]
+        public string? ConnectionString { get; set; }
+    }
+
+    public class Validator : AbstractValidator<AddOdsInstanceDerivativeRequest>
+    {
+        private readonly IGetOdsInstanceQuery _getOdsInstanceQuery;
+        private readonly string _databaseEngine;
+
+        public Validator(IGetOdsInstanceQuery getOdsInstanceQuery, IOptions<AppSettings> options)
+        {
+            _getOdsInstanceQuery = getOdsInstanceQuery;
+            _databaseEngine = options.Value.DatabaseEngine ?? throw new NotFoundException<string>("AppSettings", "DatabaseEngine");
+
+            RuleFor(m => m.DerivativeType).NotEmpty();
```

Comment:
@lchavarriaga 
OdsInstance derivative should be unique by combined key value (OdsInstanceId and DerivativeType). Need to have validation for that. Sorry found out that while testing.